src/Policy.hack

```diff
@@ -4,26 +4,131 @@
 namespace HTMLPurifier;
 use namespace HH\Lib\{C, Str};
 
+use type HTMLPurifier\Enums\{HtmlTags, HtmlAttributes};
+
+type html_input_t = shape(
+	'dirty_html' => string,
+	'policy' => HTMLPurifier_Policy,
+);
+
+/*
+* Default policy constant. This is the list of default allowed tags and
+* attributes when calling `HTMLPurifier_Policy::fromEmpty()`
+ */
+const dict<HtmlTags, keyset<HtmlAttributes>> HTML_SANITIZER_DEFAULT_POLICY = dict[
+	HtmlTags::P => keyset[],
+	HtmlTags::B => keyset[],
+	HtmlTags::I => keyset[],
+];
+
+final class HTMLSanitizerException extends \Exception {}
+
 class HTMLPurifier_Policy {
-	public dict<string, vec<string>> $allowed_tags_attributes = dict[];
+	// public dict<string, vec<string>> $allowed_tags_attributes = dict[];
+
+	private dict<HtmlTags, keyset<HtmlAttributes>> $html_purifier_policy = dict[];
+
+	/**
+	 * Construct policy class from set of predefined policy allowlist
+	 *
+	 * @param html_purifier_policy_from_t $policyFrom select from predefined policy
+	 * @return HTMLPurifier_Policy the current HTMLPurifier_Policy object
+	 */
+	public function __construct(dict<HtmlTags, keyset<HtmlAttributes>> $policy) {
+		foreach ($policy as $allowed_tag => $allowed_attributes) {
+			$this->addAllowedTagWithAttributes($allowed_tag, $allowed_attributes);
+		}
+	}
+
+	/**
+	 * Use the default allowlist policy.
+	 *
+	 * @return HTMLPurifier_Policy the current HTMLPurifier_Policy object
+	 */
+	public static function fromDefault(): HTMLPurifier_Policy {
+		$html_purifier_policy = HTML_SANITIZER_DEFAULT_POLICY;
+		return new HTMLPurifier_Policy($html_purifier_policy);
+	}
+
+	/**
+	 * Use the empty allowlist policy.
+	 *
+	 * @return HTMLPurifier_Policy the current HTMLPurifier_Policy object
+	 */
+	public static function fromEmpty(): HTMLPurifier_Policy {
+		return new HTMLPurifier_Policy(dict[]);
+	}
+
+	/**
+	 * Add a single allowed tag to the allowlist. This only allows the tag with
+	 * any default safe attributes that the library sets.
+	 *
+	 * @param HtmlTags $tag
+	 * @return HTMLPurifier_Policy the current HTMLPurifier_Policy object
+	 */
+	public function addAllowedTag(HtmlTags $allowed_tag): HTMLPurifier_Policy {
+		// Skip if we already have the same tag in the policy
```

Comment:
since we `throw` and don't actually skip, let's remove this comment.

Reply:
I'm wondering if it makes sense to throw here now that I'm looking at it because it doesn't seem like the most user-friendly behavior. I feel like we could just actually skip it to be honest (instead of throwing an exception that needs to be handled).